ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/FlatEventsObserver.scala

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.daml.ledger.api.benchtool.util.ObserverWithResult
+import com.daml.ledger.api.v1.transaction_service.GetTransactionsResponse
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.concurrent.Future
+
+object FlatEventsObserver {
+  def apply(expectedTemplateNames: Set[String]): FlatEventsObserver = new FlatEventsObserver(
+    logger = LoggerFactory.getLogger(getClass),
+    expectedTemplateNames = expectedTemplateNames,
+  )
+}
+
+/** Collects information about create and exercise events.
+  */
+class FlatEventsObserver(expectedTemplateNames: Set[String], logger: Logger)
+    extends ObserverWithResult[GetTransactionsResponse, ObservedEvents](logger) {
+
+  private val createEvents = collection.mutable.ArrayBuffer[ObservedCreateEvent]()
+  private val exerciseEvents = collection.mutable.ArrayBuffer[ObservedExerciseEvent]()
+
+  override def streamName: String = "dummy-stream-name"
```

Comment:
Is this a used in unit test or is this dummy override an oversight?

Reply:
That's used just in tests so I think it's fine. 
I also plan to move these observers to a test-lib target